### PR TITLE
DEP: Fix post-deprecation warnings

### PR DESCRIPTION
**Pull request by larsoner**
Should fix https://app.circleci.com/pipelines/github/mne-tools/mne-python/19407/workflows/351149c0-a8b7-48e9-b87a-15e299bffbdb/jobs/55471

### CODE REVIEW

File: mne/viz/epochs.py [hunk 1]
```diff
@@ -927,7 +927,11 @@ def plot_epochs(
         if not hasattr(event_id, "keys"):
             event_id = dict()
         # TODO: when min py=3.9, change to `epochs.event_id | event_id`
-        event_id = dict(**epochs.event_id, **event_id)
+        # Let passed-in event_id take precedence (and avoid repeated kwargs)
```
Comment (drammock) [inline mne/viz/epochs.py:930]: Does `event_id = epochs.event_id.copy().update(event_id)` not work? The comprehension isn't pretty but I guess if it's the only way that works in 3.8 there's little we can do.

**Review by drammock — APPROVED** [review]
Just one comment below. Thanks for cleaning up after me

File: mne/viz/epochs.py (hunk 1 above)
Comment (larsoner) [reply mne/viz/epochs.py:930]: Yeah that should work, I'll push that

Comment (larsoner) [reply mne/viz/epochs.py:929]: Turned out not to be as trivial as I thought because of some corner cases found thanks to TDD

**Pull request merged**